src/components/common/TeacherFooter.tsx

```diff
@@ -0,0 +1,97 @@
+import { useNavigate } from "react-router-dom";
+import { useRecoilState } from "recoil";
+import styled from "styled-components";
+import {
+  CalendarActiveTeacherFooterIc,
+  CalendarTeacherFooterIc,
+  ClassManagingActiveTeacherFooterIc,
+  ClassManagingTeacherFooterIc,
+  HomeActiveTeacherFooterIc,
+  HomeTeacherFooterIc,
+  MyActiveTeacherFooterIc,
+  MyTeacherFooterIc,
+} from "../../assets";
+import { teacherFooterCategory } from "../../atom/teacherFooterCategory";
+import { TEACHER_FOOTER_CATEGORY } from "../../core/teacherFooter";
+import { TeacherFooterType } from "../../type/teacherFooterType";
+
+export default function TeacherFooter() {
+  const [teacherFooterList, setTeacherFooterList] = useRecoilState<TeacherFooterType[]>(teacherFooterCategory);
+  const navigate = useNavigate();
+
+  function showTeacherFooterIcon(category: string, isMoved: boolean) {
+    switch (category) {
+      case TEACHER_FOOTER_CATEGORY.home:
+        return isMoved ? <HomeActiveTeacherFooterIc /> : <HomeTeacherFooterIc />;
+      case TEACHER_FOOTER_CATEGORY.calendar:
+        return isMoved ? <CalendarActiveTeacherFooterIc /> : <CalendarTeacherFooterIc />;
+      case TEACHER_FOOTER_CATEGORY.classManaging:
+        return isMoved ? <ClassManagingActiveTeacherFooterIc /> : <ClassManagingTeacherFooterIc />;
+      case TEACHER_FOOTER_CATEGORY.my:
+        return isMoved ? <MyActiveTeacherFooterIc /> : <MyTeacherFooterIc />;
+      default:
+        return;
+    }
+  }
+
+  function handleMoveToPage(category: string) {
+    setTeacherFooterList(
+      teacherFooterList.map((list) =>
+        list.category === category ? { ...list, isMoved: true } : { ...list, isMoved: false },
+      ),
+    );
+    switch (category) {
+      case TEACHER_FOOTER_CATEGORY.home:
+        navigate("/");
+        break;
+      case TEACHER_FOOTER_CATEGORY.calendar:
+        navigate("/schedule");
+        break;
+      case TEACHER_FOOTER_CATEGORY.classManaging:
+        navigate("/check-lesson");
+        break;
+      case TEACHER_FOOTER_CATEGORY.my:
+        navigate("/mypage");
+        break;
+      default:
+        break;
+    }
+  }
+
+  return (
+    <TeacherFooterWrapper>
+      {teacherFooterList.map(({ id, category, isMoved }) => (
+        <Icon key={id} onClick={() => handleMoveToPage(category)}>
```

Comment:
근데 이거 아까 header PR에는
```suggestion
        <Icon key={id} onClick={handleMoveToPage(category)}>
```
이런 식으로 되어있었는데 제가 예에전에 코드리뷰 받았을 때는 `onClick={handleMoveToPage(category)}` 이런식으로 써주면 클릭이벤트의 발생 유무와 상관없이 바로 실행된다고,,,? 그래서 `onClick={()=>handleMoveToPage(category)}` 이렇게 써주라고 들었는데 지금 또 그 정보의 출처를 찾으려고 하니,,,잘 안보이네요
뭐가 맞는 걸까나 [이런 게 있긴한데 개인 블로그라,,,](https://velog.io/@xmun74/TIL-onClick%EC%97%90%EC%84%9C-%ED%99%94%EC%82%B4%ED%91%9C%ED%95%A8%EC%88%98)

Reply:
헐 그렇군요.... 근데 onClick={handleMoveToPage(category)} 이렇게 쓰면 에러가 나던데.......뭘까.........